compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -195,7 +195,8 @@ private BPackageSymbol definePackage(PackageID packageId,
             this.env = prevEnv;
             return pkgSymbol;
         } catch (Throwable e) {
-            throw new BLangCompilerException(e.getMessage(), e);
+            throw new BLangCompilerException("Error while loading the bir. " +
+                    (e.getMessage() != null ? e.getMessage() : ""), e);
```

Comment:
Since this is a `BLangCompilerException`, this will get printed with the `error:` prefix. So starting the message with `Error` may look like redundant. Also, shall we separate the messages for the 2 conditions (where error message is `null` and not `null`)? Something like,
- if the error message is there, `"failed to load the module '" + packageId.toString() + "' from its BIR due to: " + e.getMessage()`
- if the error message is `null`, `"failed to load the module '" + packageId.toString() + "' from its BIR"`

Also wondering if that 2nd case should include an additional hint like `Please ensure that the module version used is compatible with the language version in use` 🤔 

@MaryamZi thoughts?

Reply:
+1 to start with "failed to".

Re: the actual error though, is this purely due to not being able to load the module from the BIR? Or can it also happen maybe due to the BIR not being available for example?

 
> Also wondering if that 2nd case should include an additional hint like `Please ensure that the module version used is compatible with the language version in use` 

Same concern as above, is this the only reason that can cause the message to be null?

Reply:
If the BIR is not available at all, it'll fail when trying to resolve the import right? Saying that it could not resolve the module? @azinneera please confirm.

The thing with the second one is a bit tricky. Version incompatibility may not be the only cause for NPEs. It could be a bug in the compiler too. But the common scenario I've seen for NPEs is the version incompatibility. Hence the suggestion to add that hint. The desirable situation would be to be able to distinguish between the error cases: failure due to version incompatibility (give an error message to the user), compiler bugs (crash with bad-sad). But can't really do that with the current design.

Reply:
We load the BIR only if it exists in the file system. Else we use the bala files to compile. https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java#L246. So this will only when the BIR is available, AFAIU.

Reply:
We load the BIR only if it exists in the file system. Else, we use the bala files to compile. https://github.com/ballerina-platform/ballerina-lang/blob/8be01a443677993bb55965af19b8d1843f474f6b/compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java#L246. 
https://github.com/ballerina-platform/ballerina-lang/blob/8be01a443677993bb55965af19b8d1843f474f6b/compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java#L63-L75
So this will only happen when the BIR is available, AFAIU.

Reply:
Thanks for the inputs. Shall we stick to the below suggestions since the versioning issue cannot be distinguished? Even though the hint will address the common case, it might be misleading for the rest. @pubudu91 @MaryamZi 
> if the error message is there, "failed to load the module '" + packageId.toString() + "' from its BIR due to: " + e.getMessage()
> if the error message is null, "failed to load the module '" + packageId.toString() + "' from its BIR"

Reply:
+1